libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,235 @@
+                new ResourceExplorer()
+                    .AddFolder(applicationRoot)
+                    .RegisterType<OnQnAMatch>(OnQnAMatch.Kind));
+            
+            // Runtime set up
+            services.AddBotRuntimeSkills(runtimeSettings.Skills);
+            services.AddBotRuntimeStorage(configuration, runtimeSettings);
+            services.AddBotRuntimeTelemetry(runtimeSettings.Telemetry);
+            services.AddBotRuntimeTranscriptLogging(configuration, runtimeSettings.Features);
+            services.AddBotRuntimeFeatures(runtimeSettings.Features);
+            services.AddBotRuntimePlugins(configuration, runtimeSettings);
+            services.AddBotRuntimeAdapters(runtimeSettings);
+        }
+
+        internal static void AddBotRuntimeSkills(this IServiceCollection services, SkillSettings skillSettings)
+        {
+            services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new AllowedCallersClaimsValidator(skillSettings?.AllowedCallers) });
+            services.AddSingleton<SkillConversationIdFactoryBase, SkillConversationIdFactory>();
+            services.AddHttpClient<BotFrameworkClient, SkillHttpClient>();
+            services.AddSingleton<ChannelServiceHandler, SkillHandler>();
+        }
+
+        internal static void AddBotRuntimeStorage(this IServiceCollection services, IConfiguration configuration, RuntimeSettings runtimeSettings)
+        {
+            services.AddSingleton<UserState>();
+            services.AddSingleton<ConversationState>();
+
+            // Cosmosdb
+            if (runtimeSettings?.Storage == typeof(CosmosDbPartitionedStorage).Name)
+            {
+                var cosmosDbOptions = configuration?.GetSection(typeof(CosmosDbPartitionedStorage).Name).Get<CosmosDbPartitionedStorageOptions>();
+                services.AddSingleton<IStorage>(sp => new CosmosDbPartitionedStorage(cosmosDbOptions));
+            }
+
+            // Blob
+            else if (runtimeSettings?.Storage == typeof(BlobsStorage).Name)
+            {
+                var blobOptions = configuration?.GetSection(typeof(BlobsStorage).Name).Get<BlobsStorageSettings>();
+                services.AddSingleton<IStorage>(sp => new BlobsStorage(blobOptions?.ConnectionString, blobOptions?.ContainerName));
+            }
+
+            // Default
+            else
+            {
+                // If no storage is configured, default to memory storage
+                services.AddSingleton<IStorage, MemoryStorage>();
+            }
+        }
+
+        internal static void AddBotRuntimePlugins(this IServiceCollection services, IConfiguration configuration, RuntimeSettings runtimeSettings)
+        {
+            using (IServiceScope serviceScope = services.BuildServiceProvider().CreateScope())
+            {
+                var pluginEnumenator = serviceScope.ServiceProvider.GetService<IBotPluginEnumerator>() ?? new AssemblyBotPluginEnumerator(AssemblyLoadContext.Default);
```

Comment:
Typo pluginEnumenator

Reply:
Dotnet uses Enumerator in system.collections: https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator?view=net-5.0

English is not my first language and I'm not sure what the right word is so I'm staying consistent with them. Let me know if you think othewise,.

Reply:
I thought you were proposing enumenator was the correct one. Understood now :)